Dockerfile

```diff
@@ -1,31 +1,81 @@
-FROM jenkins
+FROM jenkins:2.32.2
 
 USER root
 
+RUN apt-get -y update
+
 # Add the s6 overlay.
-ENV S6_VERSION v1.16.0.0
+ENV S6_VERSION v1.19.1.1
 RUN curl -L "https://github.com/just-containers/s6-overlay/releases/download/$S6_VERSION/s6-overlay-amd64.tar.gz" | \
     tar xzvf - -C /
 ENV S6_BEHAVIOUR_IF_STAGE2_FAILS 1
 
 # Download confd.
-ENV CONFD_VERSION 0.10.0
+ENV CONFD_VERSION 0.11.0
 RUN curl -L "https://github.com/kelseyhightower/confd/releases/download/v$CONFD_VERSION/confd-$CONFD_VERSION-linux-amd64" > /usr/bin/confd && \
     chmod +x /usr/bin/confd
 ENV CONFD_OPTS '--backend=env --onetime'
 
 COPY root /
-COPY plugins.txt /usr/share/jenkins/plugins.txt
-
-RUN /usr/local/bin/plugins.sh /usr/share/jenkins/plugins.txt
 
 # Add docker binaries directly
-RUN apt-key adv --keyserver hkp://p80.pool.sks-keyservers.net:80 --recv-keys 58118E89F3A912897C070ADBF76221572C52609D
-RUN curl -L http://ftp.us.debian.org/debian/pool/main/a/apt/apt-transport-https_1.0.9.8.3_amd64.deb > /tmp/apt-transport-https_1.0.9.8.3_amd64.deb && dpkg -i /tmp/apt-transport-https_1.0.9.8.3_amd64.deb
+RUN apt-get -y install \
+        apt-transport-https \
+        software-properties-common
+RUN curl -fsSL https://apt.dockerproject.org/gpg | apt-key add -
+RUN add-apt-repository \
+       "deb https://apt.dockerproject.org/repo/ \
+       debian-jessie \
+       main"
 RUN apt-get update
 RUN apt-cache policy docker-engine
-RUN apt-get -y install docker-engine=1.9.1-0~jessie
-RUN curl -L https://github.com/docker/compose/releases/download/1.5.2/docker-compose-`uname -s`-`uname -m` > /usr/bin/docker-compose && chmod +x /usr/bin/docker-compose
+RUN apt-get -y install docker-engine=1.13.1-0~debian-jessie
+RUN curl -L https://github.com/docker/compose/releases/download/1.11.2/docker-compose-`uname -s`-`uname -m` > /usr/bin/docker-compose && \
+      chmod +x /usr/bin/docker-compose
+
+RUN install-plugins.sh \
```

Comment:
These plugins capture a lot of what makes Jenkins2, well Jenkins2. A lot of the usability/functionality changes are in the default plugins that are now installed.  The workflow and folders, etc are part of the default Jenkins 2 install, so I just added them explicitly since we skip the initial installation wizard that ads them.